plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,381 @@
+    ) -> Tuple[str, int]:
+        """
+        Takes custom config from CPS and allows the specification of a new start time for alerts,
+        and allows the limit to be customised.
+
+        :param custom_config: custom_config for the plugin
+        :param now: datetime representing 'now'
+        :param saved_state: existing state of the integration
+        :return: string formatted time to apply to our alert queries, and limit integer to set how many
+        alerts to fetch per run.
+        """
+        state = saved_state.copy()
+
+        # set the alert limit from CPS if it exists, otherwise default to ALERT_LIMIT
+        alert_limit = custom_config.get("alert_limit", ALERT_LIMIT)
+
+        saved_time = state.get(LAST_ALERT_TIME)
+
+        log_msg = ""
+        if not saved_time:
+            log_msg += "No previous alert time within state. "
+            custom_timings = custom_config.get(LAST_ALERT_TIME, {})
+            custom_date = custom_timings.get("date")
+            custom_hours = custom_timings.get("hours", DEFAULT_LOOKBACK_HOURS)
+            start = datetime(**custom_date) if custom_date else (now - timedelta(hours=custom_hours))
+            state[LAST_ALERT_TIME] = start.strftime(TIME_FORMAT)
+        else:
+            # check if we have held the TS beyond our max lookback
+            lookback_days = custom_config.get(f"{LAST_ALERT_TIME}_days", MAX_LOOKBACK_DAYS)
+            default_date_lookback = now - timedelta(days=lookback_days)  # if not passed from CPS create on the fly
+            custom_lookback = custom_config.get(f"max_{LAST_ALERT_TIME}", {})
+            comparison_date = datetime(**custom_lookback) if custom_lookback else default_date_lookback
+            comparison_date = comparison_date.replace(tzinfo=timezone.utc).strftime(TIME_FORMAT)
+            if comparison_date > saved_time:
+                self.logger.info(f"Saved time ({saved_time}) exceeds cut off, moving to ({comparison_date}).")
+                state[LAST_ALERT_TIME] = comparison_date
+                # pop state held time filters if they exist, incase customer has paused integration when paginating
+                state = self._drop_pagination_state(state)
+
+        start_time = state.get(LAST_ALERT_TIME)
+        print(f"{start_time = }")
+        if not state.get(FROM_TIME_FILTER):
+            self.logger.info(f"{log_msg}Applying the following start time='{start_time}'. Limit={alert_limit}.")
+        else:
+            # we're paginating, this will be the value used to query for alerts
+            self.logger.info(
+                f"{log_msg}Applying the following start time='{state.get(FROM_TIME_FILTER)}'. " f"Limit={alert_limit}"
+            )
+
+        return start_time, alert_limit
+
+    ###########################
+    # Handle Pagination
+    ###########################
+    def _drop_pagination_state(self, state: Dict[str, str]) -> Dict[str, str]:
```

Comment:
I think we also need to drop the last_search_from and last_search_to to allow to prevent continuous pagination. The API accepts that we can search past the total count.

Reply:
Looking into whether this method is needed or not.